module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -146,17 +146,17 @@ func (k Keeper) verifyERC20DeployedEvent(ctx sdk.Context, event *types.ERC20Depl
 }
 
 func verifyERC20Token(metadata banktypes.Metadata, event *types.ERC20DeployedEvent) error {
-	if event.Erc20Name != metadata.Display {
+	if event.Erc20Name != metadata.Name {
```

Comment:
Aren't these checks incompatible with the values from the updated `DenomToERC20ParamsResponse`, since it will fall back to `Display` if these values are not present?

Reply:
This function is called to verify new cosmos-erc20 tokens deployed by the bridge. It means that after the upgrade, all the coins needs to have name and symbol in their metadatas.

For the existing one (those who are deployed after the upgrade), they may not have name or symbol in their metadata, that is why we fallback to display in `DenomToERC20ParamsResponse`